pycbc/waveform/compress.py

```diff
@@ -459,15 +474,35 @@ def fd_decompress(amp, phase, sample_frequencies, out=None, df=None,
         If out was provided, writes to that array. Otherwise, a new
         FrequencySeries with the decompressed waveform.
     """
-    precision = _precision_map[sample_frequencies.dtype.name]
-    if _precision_map[amp.dtype.name] != precision or \
-            _precision_map[phase.dtype.name] != precision:
-        raise ValueError("amp, phase, and sample_points must all have the "
-            "same precision")
+    precision = _precision_map[phase_freq.dtype.name]
+    if interpolation == "linear":
+        if amp_freq != None:
+            raise ValueError("for inline_linear_single_freq decompression"
+                     "amp_freq should be set to None as it should be the"
+                     " same as phase_freq")
+        sample_frequencies = phase_freq
+        if _precision_map[amp.dtype.name] != precision or \
+               _precision_map[phase.dtype.name] != precision:
+           raise ValueError("amp, phase, and phase_freq must"
+                             " all have the same precision")
```

Comment:
__[flake8]__
* [E127] continuation line over-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->